tests/config_parse_test.cpp

```diff
@@ -283,3 +285,129 @@ TEST(ConfigParse, ConfigRoot) {
   EXPECT_NO_THROW(flexi_cfg::Parser::parse(
       std::filesystem::path("config_root/test/config_example_base.cfg"), baseDir()));
 }
+
+TEST(ConfigVisitor, JsonConfigVisitor) {
```

Comment:
Test coverage seems kind of light here. I'm not suggesting we cover the entire json spec, but maybe we can add some of the official json schema test cases here? https://github.com/json-schema-org/JSON-Schema-Test-Suite/tree/main/tests/draft2020-12

Reply:
Found a repo with actual .json test case files: https://github.com/nlohmann/json_test_data

Reply:
The goal of this test was to cover all FlexiConfig features and ensure the JSON visitors output a valid JSON for all these constructs.

Do you have an example of something we're missing in `config_example16.cfg`?

Reply:
Ah ok I misunderstood. In that case, I think we are missing:
- includes, empty lists, and overrides?
- booleans in lists/structs
- fully qualified key paths like `my_struct.var` should end up inside `my_struct` in the final json.